src/domain/synthetics/markets/createDepositTxn.ts

```diff
@@ -125,6 +127,120 @@ export async function createDepositTxn(chainId: number, signer: Signer, p: Param
+
+  const multicall = [
+    { method: "sendWnt", params: [depositVaultAddress, wntAmount] },
+    !isNativeLongDeposit && p.longTokenAmount > 0 && !p.isMarketTokenDeposit
+      ? { method: "sendTokens", params: [p.initialLongTokenAddress, depositVaultAddress, p.longTokenAmount] }
+      : undefined,
+
+    !isNativeShortDeposit && p.shortTokenAmount > 0 && !p.isMarketTokenDeposit
+      ? { method: "sendTokens", params: [p.initialShortTokenAddress, depositVaultAddress, p.shortTokenAmount] }
+      : undefined,
+    p.isMarketTokenDeposit
+      ? {
+          method: "sendTokens",
+          params: [p.initialLongTokenAddress, depositVaultAddress, p.longTokenAmount],
+        }
+      : undefined,
+    {
+      method: "createGlvDeposit",
+      params: [
+        {
+          glv: p.marketTokenAddress,
+          market: p.market,
+          receiver: p.account,
+          callbackContract: ethers.ZeroAddress,
+          uiFeeReceiver: UI_FEE_RECEIVER_ACCOUNT ?? ethers.ZeroAddress,
+          initialLongToken: p.isMarketTokenDeposit ? ethers.ZeroAddress : initialLongTokenAddress,
+          initialShortToken: p.isMarketTokenDeposit ? ethers.ZeroAddress : initialShortTokenAddress,
+          longTokenSwapPath: [],
+          shortTokenSwapPath: [],
+          minGlvTokens: minGlvTokens,
+          executionFee: p.executionFee,
+          callbackGasLimit: 0n,
+          shouldUnwrapNativeToken,
+          isMarketTokenDeposit: p.isMarketTokenDeposit,
+        },
+      ],
+    },
+  ];
+
+  const encodedPayload = multicall
+    .filter(Boolean)
+    .map((call) => contract.interface.encodeFunctionData(call!.method, call!.params));
+
+  if (!p.skipSimulation) {
+    await simulateExecuteTxn(chainId, {
+      account: p.account,
+      primaryPriceOverrides: {},
+      tokensData: p.tokensData,
+      createMulticallPayload: encodedPayload,
+      method: "simulateExecuteGlvDeposit",
+      errorTitle: t`Deposit error.`,
+      value: wntAmount,
+    });
+  }
+
+  return callContract(chainId, contract, "multicall", [encodedPayload], {
+    value: wntAmount,
+    hideSentMsg: true,
+    hideSuccessMsg: true,
+    metricId: p.metricId,
+    setPendingTxns: p.setPendingTxns,
+  }).then(() => {
+    p.setPendingDeposit({
+      account: p.account,
+      marketAddress: p.marketTokenAddress,
+      initialLongTokenAddress: p.isMarketTokenDeposit ? ethers.ZeroAddress : initialLongTokenAddress,
+      initialShortTokenAddress: p.isMarketTokenDeposit ? ethers.ZeroAddress : initialShortTokenAddress,
+      longTokenSwapPath: p.longTokenSwapPath,
+      shortTokenSwapPath: p.shortTokenSwapPath,
+      minMarketTokens: p.minMarketTokens,
+      shouldUnwrapNativeToken,
+      initialLongTokenAmount: p.longTokenAmount,
+      initialShortTokenAmount: p.shortTokenAmount,
+      isGlvDeposit: true,
+      gmAddress: p.isMarketTokenDeposit ? p.initialLongTokenAddress : undefined,
```

Comment:
I think we currently have 1 file per each tx creator,
let's create `createGlvDepositTxn` and same for withdrawals and shifts?
They don't share much in common besides the name I think